src/microbe_stage/organelle_unlocks/UnlockProgress.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Stores which organelles have been unlocked by the player.
+/// </summary>
+[UseThriveSerializer]
+public class UnlockProgress
+{
+    /// <summary>
+    ///   The organelles that the player can use.
+    ///   The default value is the by default unlocked organelles (ones without a condition)
+    /// </summary>
+    [JsonProperty]
+    private readonly HashSet<OrganelleDefinition> unlockedOrganelles =
+        SimulationParameters.Instance.GetAllOrganelles().Where(organelle => organelle.UnlockConditions == null).
+            ToHashSet();
+
+    /// <summary>
+    ///   Organelles unlocked singe the last time in the editor.
+    /// </summary>
+    [JsonProperty]
+    private readonly HashSet<OrganelleDefinition> recentlyUnlocked = new();
+
+    /// <summary>
+    ///  If true, <see cref="IsUnlocked"/> will always return true
+    /// </summary>
+    [JsonProperty]
+    public bool UnlockAll { get; set; }
+
+    /// <summary>
+    ///   Unlock an organelle, returning true if this is the first time it has been unlocked.
+    /// </summary>
+    public bool UnlockOrganelle(OrganelleDefinition organelle, GameProperties game)
+    {
+        if (organelle.UnlockConditions == null || game.FreeBuild || UnlockAll)
+            return false;
+
+        var firstTimeUnlocking = unlockedOrganelles.Add(organelle);
+
+        if (firstTimeUnlocking)
+            recentlyUnlocked.Add(organelle);
+
+        return firstTimeUnlocking;
+    }
+
+    /// <summary>
+    ///   Is the organelle unlocked?
+    /// </summary>
+    public bool IsUnlocked(OrganelleDefinition organelle, GameProperties game, bool autoUnlock)
+    {
+        if (organelle.UnlockConditions == null || game.FreeBuild || UnlockAll)
+            return true;
+
+        if (organelle.UnlockConditions.Any(unlock => unlock.Satisfied()) && autoUnlock)
```

Comment:
```suggestion
        if (autoUnlock && organelle.UnlockConditions.Any(unlock => unlock.Satisfied()))
```
I'd swap the condition here to be this way around, that way the more expensive satisfied condition loop is not executed if auto unlock happens to be false.